server/src/main/java/org/apache/druid/server/http/DataSegmentDto.java

```diff
@@ -0,0 +1,224 @@
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import org.apache.druid.guice.annotations.PublicApi;
+import org.apache.druid.jackson.CommaListJoinDeserializer;
+import org.apache.druid.jackson.CommaListJoinSerializer;
+import org.apache.druid.timeline.CompactionState;
+import org.apache.druid.timeline.DataSegment;
+import org.apache.druid.timeline.SegmentId;
+import org.apache.druid.timeline.partition.ShardSpec;
+import org.joda.time.DateTime;
+import org.joda.time.Interval;
+
+import javax.annotation.Nullable;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * {@link DataSegment} to be transmitted externally
+ */
+@PublicApi
+public class DataSegmentDto
+{
+  private final DataSegment dataSegment;
+  private final DateTime createdDate;
+  @Nullable
+  private final DateTime usedStatusLastUpdatedDate;
+
+  public DataSegmentDto(
+      DataSegment dataSegment,
+      DateTime createdDate,
+      @Nullable DateTime usedStatusLastUpdatedDate
+  )
+  {
+    this.dataSegment = dataSegment;
+    this.createdDate = createdDate;
+    this.usedStatusLastUpdatedDate = usedStatusLastUpdatedDate;
+  }
+
+  @JsonCreator
+  public DataSegmentDto(
+      @JsonProperty("dataSource") String dataSource,
+      @JsonProperty("interval") Interval interval,
+      @JsonProperty("version") String version,
+      // use `Map` *NOT* `LoadSpec` because we want to do lazy materialization to prevent dependency pollution
+      @JsonProperty("loadSpec") @Nullable Map<String, Object> loadSpec,
+      @JsonProperty("dimensions")
+      @JsonDeserialize(using = CommaListJoinDeserializer.class)
+      @Nullable
+      List<String> dimensions,
+      @JsonProperty("metrics")
+      @JsonDeserialize(using = CommaListJoinDeserializer.class)
+      @Nullable
+      List<String> metrics,
+      @JsonProperty("shardSpec") @Nullable ShardSpec shardSpec,
+      @JsonProperty("lastCompactionState") @Nullable CompactionState lastCompactionState,
+      @JsonProperty("binaryVersion") Integer binaryVersion,
+      @JsonProperty("size") long size,
+      @JsonProperty("createdDate") @Nullable DateTime createdDate,
+      @JsonProperty("createdDate") @Nullable DateTime usedStatusLastUpdatedDate,
+      @JacksonInject DataSegment.PruneSpecsHolder pruneSpecsHolder
+  )
+  {
+    this.dataSegment = new DataSegment(
+        dataSource,
+        interval,
+        version,
+        loadSpec,
+        dimensions,
+        metrics,
+        shardSpec,
+        lastCompactionState,
+        binaryVersion,
+        size,
+        pruneSpecsHolder
+    );
+    this.createdDate = createdDate;
+    this.usedStatusLastUpdatedDate = usedStatusLastUpdatedDate;
+  }
+
+  /**
+   * Get dataSource
+   *
+   * @return the dataSource
+   */
+  @JsonProperty
+  public String getDataSource()
```

Comment:
Ditto. All these getters can be from the `DataSegment` embedded inside here.

Reply:
fixed